src/sql/workbench/services/metadata/common/serverMetadataService.ts

```diff
@@ -0,0 +1,91 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import { invalidProvider } from 'sql/base/common/errors';
+import { IConnectionManagementService, IConnectionParams } from 'sql/platform/connection/common/connectionManagement';
+import { IServerMetadataService } from 'sql/workbench/services/metadata/common/interfaces';
+import { Disposable } from 'vs/base/common/lifecycle';
+
+export class ServerMetadataService extends Disposable implements IServerMetadataService {
+	public _serviceBrand: undefined;
+	private _providers = new Map<string, azdata.metadata.ServerMetadataProvider>();
+
+	constructor(
+		@IConnectionManagementService private readonly _connectionManagementService: IConnectionManagementService
+	) {
+		super();
+
+		this._register(this._connectionManagementService.onConnect(async (e: IConnectionParams) => {
+			const ownerUri = e.connectionUri;
+			await this.generateServerMetadata(ownerUri);
```

Comment:
This seems expensive to do on every connect, we probably shouldn't be doing it until it actually is going to be used.